File: jvm/sargon-android/src/main/java/com/radixdlt/sargon/extensions/AppearanceId.kt
package com.radixdlt.sargon.extensions

import com.radixdlt.sargon.AppearanceId
import com.radixdlt.sargon.appearanceIdsAll
import com.radixdlt.sargon.newAppearanceId

@Throws(SargonException::class)
fun AppearanceId.Companion.init(validating: UByte) = newAppearanceId(validating = validating)

fun AppearanceId.Companion.all() = appearanceIdsAll()

File: crates/sargon-uniffi/src/core/types/appearance_id.rs
use crate::prelude::*;
use sargon::AppearanceID as InternalAppearanceID;

#[derive(Clone, PartialEq, Eq, Hash, InternalConversion, uniffi::Record)]
pub struct AppearanceID {
    pub value: u8,
}

#[uniffi::export]
pub fn new_appearance_id(validating: u8) -> Result<AppearanceID> {
    InternalAppearanceID::new(validating).into_result()
}

#[uniffi::export]
pub fn new_appearance_id_from_number_of_accounts_on_network(
    count: u64,
) -> AppearanceID {
    InternalAppearanceID::from_number_of_accounts_on_network(count as usize)
        .into()
}

#[uniffi::export]
pub fn new_appearance_id_sample() -> AppearanceID {
    InternalAppearanceID::sample().into()
}

#[uniffi::export]
pub fn new_appearance_id_sample_other() -> AppearanceID {
    InternalAppearanceID::sample_other().into()
}

#[uniffi::export]
pub fn appearance_ids_all() -> Vec<AppearanceID> {
    InternalAppearanceID::all().into_type()
}
